dft/src/radix_2_bowers.rs

```diff
@@ -77,80 +75,54 @@ impl<F: TwoAdicField> TwoAdicSubgroupDft<F> for Radix2Bowers {
 
 /// Executes the Bowers G network. This is like a DFT, except it assumes the input is in
 /// bit-reversed order.
-fn bowers_g<F: TwoAdicField>(mat: &mut RowMajorMatrixViewMut<'_, F>) {
+fn bowers_g<F: TwoAdicField>(mat: &mut RowMajorMatrixViewMut<F>) {
     let h = mat.height();
     let log_h = log2_strict_usize(h);
 
     let root = F::two_adic_generator(log_h);
-    let mut twiddles: Vec<F> = root.powers().take(h / 2).collect();
+    let mut twiddles: Vec<_> = root.powers().take(h / 2).map(DifButterfly).collect();
     reverse_slice_index_bits(&mut twiddles);
 
     let log_h = log2_strict_usize(mat.height());
     for log_half_block_size in 0..log_h {
-        bowers_g_layer(mat, log_half_block_size, &twiddles);
+        // bowers_g_layer(mat, log_half_block_size, &twiddles);
+        par_chunks_bowers(mat, 1 << log_half_block_size, &twiddles)
     }
 }
 
 /// Executes the Bowers G^T network. This is like an inverse DFT, except we skip rescaling by
 /// 1/height, and the output is bit-reversed.
-fn bowers_g_t<F: TwoAdicField>(mat: &mut RowMajorMatrixViewMut<'_, F>) {
+fn bowers_g_t<F: TwoAdicField>(mat: &mut RowMajorMatrixViewMut<F>) {
     let h = mat.height();
     let log_h = log2_strict_usize(h);
 
     let root_inv = F::two_adic_generator(log_h).inverse();
-    let mut twiddles: Vec<F> = root_inv.powers().take(h / 2).collect();
+    let mut twiddles: Vec<_> = root_inv.powers().take(h / 2).map(DitButterfly).collect();
     reverse_slice_index_bits(&mut twiddles);
 
     let log_h = log2_strict_usize(mat.height());
     for log_half_block_size in (0..log_h).rev() {
-        bowers_g_t_layer(mat, log_half_block_size, &twiddles);
+        par_chunks_bowers(mat, 1 << log_half_block_size, &twiddles)
     }
 }
 
-/// One layer of a Bowers G network. Equivalent to `bowers_g_t_layer` except for the butterfly.
-fn bowers_g_layer<F: Field>(
-    mat: &mut RowMajorMatrixViewMut<'_, F>,
-    log_half_block_size: usize,
-    twiddles: &[F],
-) {
-    let half_block_size = 1 << log_half_block_size;
-    let width = mat.width();
-    par_chunks_bowers(mat, width, half_block_size, twiddles, dif_butterfly_on_rows)
-}
-
-/// One layer of a Bowers G^T network. Equivalent to `bowers_g_layer` except for the butterfly.
-fn bowers_g_t_layer<F: Field>(
-    mat: &mut RowMajorMatrixViewMut<'_, F>,
-    log_half_block_size: usize,
-    twiddles: &[F],
-) {
-    let half_block_size = 1 << log_half_block_size;
-    let width = mat.width();
-    par_chunks_bowers(mat, width, half_block_size, twiddles, dit_butterfly_on_rows)
-}
-
-fn par_chunks_bowers<F: Field, Fun>(
-    mat: &mut RowMajorMatrixViewMut<'_, F>,
-    width: usize,
+fn par_chunks_bowers<F: Field, B: Butterfly<F>>(
```

Comment:
nit: we could call this something like `butterfly_layer` now that it's generic to the point there's no Bowers logic (which I guess it was already, just more obvious after your refactor)